app/controllers/provider/admin/authentication_providers_controller.rb

```diff
@@ -97,9 +97,11 @@ def authorize_authentication_provider(action = action_name)
                      token_url authorize_url user_info_url identifier_key
                      username_key trust_email].freeze
 
-  def update_params
-    params.require(:authentication_provider).permit(UPDATE_PARAMS)
-  end
+UPDATE_PARAM_DEFAULTS = { branding_state_event: 'custom_brand' }.freeze
```

Comment:
I see several problems with this.

- This shouldn't happen only for `update`, but also for `create`.

- This is business logic, so it shouldn't be in the controller.

- This controller is for authentication providers in general, but this `branding_state_event` only exists for `Github`:
![image](https://user-images.githubusercontent.com/11318903/91980935-58a67480-ed28-11ea-86ff-b70bc72f5add.png)
https://github.com/3scale/porta/blob/2346d59b412e27506980bc2a78b09437d988b515/app/models/authentication_provider/github.rb#L19-L21
So doing an `update` with this controller for another authentication provider will fail.